sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/core/lang/MigrationFile.kt

```diff
@@ -11,7 +11,7 @@ class MigrationFile(
 ) : SqlDelightFile(viewProvider, MigrationLanguage) {
   val version: Int by lazy {
     name.substringBeforeLast(".${fileType.EXTENSION}")
-        .filter { it in '0'..'9' }.toInt()
+        .filter { it in '0'..'9' }.toIntOrNull() ?: 0
```

Comment:
This logic is 'flawed'. Been meaning to make a PR but didn't get around.

`22_acttivity2.sqm` will yield in a 222 version which is probably not desired.

Reply:
Not sure how we could protect against that case. There’s already a check for migration gaps so if the naming scheme causes that the verify task should fail

Reply:
The underscore could terminate.

`name.substringBeforeLast(".${fileType.EXTENSION}")..filter { it in '0'..'9' }.takeUntil { it == '_' }.toIntOrNull() ?: 0`

Reply:
on cash they do the opposite so i don't think that would work `incident_v0010.sqm`

Reply:
i think
a) give people the ability to supply a regex for finding the version number (yuck)
b) throw errors if theres multiple separate sequences of numbers and force people to do `22_activity_two` or something

Reply:
Definitely not _a_. I think we can cover 99.9% of cases with a good regex. The other 0.1% can rename.

Reply:
My migration files look like this:

`<number>_<text>.sqm`

And ideally, I also want to place numbers in the text section.